packages/core/src/components/DateField/DateField.jsx

```diff
@@ -23,10 +23,12 @@ export class DateField extends React.PureComponent {
 
   formatDate() {
     if (this.props.dateFormatter) {
+      const yearLimit = this.props.yearLimit || 4;
+
       const values = {
-        month: this.monthInput.value,
-        day: this.dayInput.value,
-        year: this.yearInput.value
+        month: this.monthInput.value.replace(/\D+/, '').substring(0, 2),
```

Comment:
What do you think about moving this logic one level up, so from `formatDate` to `handleChange`? In order to do this, we would probably need to add a separate param to `handleChange` for the num of characters allowed (either 2 or 4)

This way if someone tries to type a third character into the month field, the `onChange` prop wont be fired at all.